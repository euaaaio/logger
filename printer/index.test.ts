import { afterEach, beforeAll, expect, it, vi } from 'vitest'

import { format } from '../test/index.js'
import { group, groupEnd, log } from './index.js'

let out = ''
let groups = 0

beforeAll(() => {
  vi.spyOn(console, 'groupCollapsed').mockImplementation((...args: any[]) => {
    console.log(...args)
    groups += 1
  })
  vi.spyOn(console, 'log').mockImplementation((...args: any[]) => {
    if (out === '') out += '\n'
    out += format(groups, ['fetch'], ...args) + '\n'
  })
  vi.spyOn(console, 'groupEnd').mockImplementation(() => {
    groups -= 1
  })
})

afterEach(() => {
  out = ''
  groups = 0
})

it('prints custom logs', () => {
  let FETCH_TYPE = {
    color: '#510080',
    name: 'fetch'
  }

  group({
    logo: true,
    message: [
      ['bold', 'Profile'],
      ['regular', 'store is trying to get new values']
    ],
    type: FETCH_TYPE
  })
  log({
    message: [
      ['bold', 'Profile'],
      ['regular', 'store received new values']
    ],
    type: FETCH_TYPE
  })
  groupEnd()

  expect(out).toMatchSnapshot()
})
