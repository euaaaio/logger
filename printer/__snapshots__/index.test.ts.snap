// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`prints custom logs 1`] = `
"
Nano Stores fetch: Profile store is trying to get new values
  fetch: Profile store received new values
"
`;
